src/adr/0041-james-servers-offering.md

```diff
@@ -0,0 +1,109 @@
+
+ - **Spring server** (`server/app`): barely maintained, Spring server exposes configuration allowing 
+class level overides, resulting in untested componant combinaison leading to unexpected runtime errors. 
+Its usage requires configuration to adapt it to the user use cases, wich can be 
+not obvious for people with limited knowledge/resources, and can discourage self hosting usages.
+
+Guice servers includes:
+
+ - **Cassandra - guice** (`server/container/guice/cassandra-guice`) and its LDAP counter-part
+(`server/container/guice/cassandra-guice-ldap`): about to be deprecated, this guice server relies on
+a Cassandra & ElasticSearch database without achieving distribution (as it misses a messaging component).
+
+ - **Distributed James** (`server/container/guice/cassandra-rabbitmq-guice`) and its LDAP counter part
+(`server/container/guice/cassandra-rabbitmq-ldap-guice`) enabled to achieve the distributed use case. This
+server furthermore is extensible, and ships many extra features by default. It is however hard to operate.
+
+ - **Guice memory** (`server/container/guice/memory-guice`) ships a server based on memory. Easy to boot, 
+without external dependencies, we mostly rely on it for testing purposes.
+
+ - **JPA Guice** (`server/container/guice/jpa-guice`) ships an extensible server based on JPA storage. With
+a default configuration using Derby embedded database, this server have no external dependencies by default.
+However the still large amount of configuration can make new comers confused. It currently satisfies operator 
+looking for self hosting solutions as well as those looking for an extensible server.
+
+ - **JPA SMTP** (`server/container/guice/jpa-smtp`) demonstrate a Mail Transfer Agent, dedicated to mail
+processing, that do not ship the James Mailbox, as well as Mail Delivery protocols like IMAP, JMAP or POP3.
+
+We can notice that this offer is specified in technical terms, and do not directly map to identified user needs.
+
+Also the large amount of servers makes the choice harder to make for newcomers.
+
+## Decision
+
+We should deliver a subset of mail server directly adressing users needs, and brand them according to their use cases
+and not due to their technical details.
+
+In order to get a clearer offer we should first reduce the server cardinality. This can be achieved via:
+
+ - **Spring server** deprecation and removal
+ - Merging Guice servers with their LDAP variations (see [ADR 36 about guice module choosing](0036-against-use-of-conditional-statements-in-guice-modules.md))
+ - **Cassandra-guice** deprecation and removal
+
+We should then rename our servers in order to match the use cases:
+
+ - **Basic server** addresses the self-hosting use case. It will be based on current *jpa-guice* server but with the clear intent to 
+lower adoption barriers. It exposes limited configurations in order not to confuse the user. Configuration and documentation is centered 
```

Comment:
I don't agree about "It exposes limited configurations in order not to confuse the user". I would really prefer a solution where "it requires very few configuration to start using it".
It would prevent the need for a switch to another server once people feel limited by the arbitrary feature selection we make.

Reply:
agree with Matthieu, I would also prefer such a solution

Reply:
Is it possible to have both? Meaning: by default a configuration is not shown at all to the Basic Operator, and a default is used without being exposed. However, if they want to config, then they can add it.

What I would like to avoid is showing something that will just cause more pain and a larger barrier to entry.

I think @chibenwa is right about focusing on use cases, and @mbaechler and @rouazana are talking about the implementation. Because we're talking about different things, it should be possible to do both.

The driver should be the use case, though. :)

Reply:
@rouazana did you writte this comment ( https://github.com/apache/james-project/pull/222#discussion_r443544577 ) before or after https://github.com/apache/james-project/pull/222#pullrequestreview-434899849

Which one better reflects your position?

Reply:
as far as the basic server can be configured to be an advanced server, I'm fine :)

Reply:
I think we all agree but fail at expressing ourselves (: